File: Second Year/Semester 3/Computer Networks/Laboratory 2/Test 1/Exercise 2/tcp_client.py
import socket
from time import sleep

IP = "192.168.0.17"
PORT = 7777

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((IP, PORT))

matrix = [[""]*3, [""]*3, [""]*3]

while True:
    i = input("Enter the row number: ")
    j = input("Enter the column number: ")
    matrix[int(i)][int(j)] = "X"
    print(matrix)
    
    s.send(i.encode())
    sleep(0.1)
    s.send(j.encode())

    iS = s.recv(128).decode()
    jS = s.recv(128).decode()

    iS = int(iS)
    jS = int(jS)
    
    matrix[iS][jS] = "O"
    print(matrix)

s.close()

File: Second Year/Semester 3/Computer Networks/Laboratory 2/Test 1/Exercise 2/tcp_client.php
<?php
$s = socket_create(AF_INET, SOCK_STREAM, 0);
socket_connect($s, "192.168.0.17", 7777);
 
$matrix = [
    ["", "", ""],
    ["", "", ""],
    ["", "", ""]
];

while (1) {
    $i = readline("Enter row number: ");
    $j = readline("Enter column number: ");
    $matrix[(int) $i][(int) $j] = "X";
    
    for ($a = 0; $a < 3; $a++) {
        for ($b = 0; $b < 3; $b++)
            echo $matrix[$a][$b];
        echo "\n";
    }

    socket_send($s, $i, 128, 0);
    sleep(0.1);
    socket_send($s, $j, 128, 0);

    socket_recv($s, $iS, 128, 0);
    socket_recv($s, $jS, 128, 0);

    $matrix[$iS][$jS] = "O";
    
    for ($a = 0; $a < 3; $a++) {
        for ($b = 0; $b < 3; $b++)
            echo $matrix[$a][$b];
        echo "\n";
    }
    
    socket_close($s);
}
?>
